strawberry/schema/schema_converter.py

```diff
@@ -410,13 +411,45 @@ def from_input_object(self, object_type: type) -> GraphQLInputObjectType:
             assert isinstance(graphql_object_type, GraphQLInputObjectType)  # For mypy
             return graphql_object_type
 
+        def check_one_of(value: dict[str, Any]) -> dict[str, Any]:
+            if len(value) != 1:
+                raise GraphQLError(
+                    f"OneOf Input Object '{type_name}' must specify exactly one key."
+                )
+
+            first_key, first_value = next(iter(value.items()))
+
+            if first_value is None or first_value is UNSET:
+                raise GraphQLError(
+                    f"Value for member field '{first_key}' must be non-null"
+                )
+
+            # We are populating all missing keys with `None`, so users
+            # don't have to set an explicit default for the input type
+            # The alternative is to tell them to use `UNSET`, but it looks
+            # a bit unfriendly to use
```

Comment:
would you prefer to use `UNSET` to keep the consistency? 😊

Reply:
I also think `UNSET` would be better, as `null` can sometimes have meaning

Reply:
@bellini666 just to clarify, we don't use `null`, as in we, don't show it in the schema, it's just a shortcut to prevent to do this:

```python
@strawberry.input(one_of=True)
class FooInput:
  a: str | None = strawberry.UNSET
  b: str | None = strawberry.UNSET
```

Reply:
@patrick91 ah got it! I think I also misread the original comment here

I agree that not requiring setting this to `strawberry.UNSET` is an unexpected behavior, but at the same time I can see how this would be justified by the fact that `one_of=True` is somewhat implicitly telling that the input need to receive _specific one_ argument.

I will generally choose the consistency path, so requiring to set this to `UNSET` would be my option. But I don't have literally nothing against not doing that.

Also, maybe we should reconsider having to do that in first place? I mean, GraphQL itself will accept `null` and `UNSET` for optional fields, so maybe we should default anything that doesn't define a default value to `UNSET` as long as it is optional? What do you think?

Reply:
ok! I'll make the change 😊

> Also, maybe we should reconsider having to do that in first place for strawberry in general? I mean, GraphQL itself will accept null and UNSET for optional fields, so maybe we should default anything that doesn't define a default value to UNSET as long as it is optional? What do you think?

mmh, this is probably a big breaking change, maybe we can discuss it on a call? we could make this configurable, but I think I need to think about it more 😊

Reply:
> I will generally choose the consistency path, so requiring to set this to UNSET would be my option. But I don't have literally nothing against not doing that.

+1 on consistency but I don't feel too strongly about it here because "the fact that `one_of=True` is somewhat implicitly telling that the input need to receive specific one argument."

> Also, maybe we should reconsider having to do that in first place for strawberry in general? I mean, GraphQL itself will accept null and UNSET for optional fields, so maybe we should default anything that doesn't define a default value to UNSET as long as it is optional? What do you think?

Agreed that if an argument is optional I would expect it automatically to default to UNSET, but I also understand this is a big change...

I don't remember what actually happens if you don't set the default to UNSET... I think it generates the schema fine but requires clients to explicitly send `null`? I can't remember.

Reply:
@bradleyoesch sorry for the late reply

> I don't remember what actually happens if you don't set the default to UNSET... I think it generates the schema fine but requires clients to explicitly send null? I can't remember.

you can't send explicit null (it's forbidden by the specification), the issue you get is that the `__init__` function will complain because we didn't send all the arguments